### PR TITLE
feat: dashboard tabs

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR adds tabs in the dashboard with data and routing to
- "Earn balances" (filtered opportunities with a positive/active earn value)
- "Rewards Balances" (filtered opportunities with a positive claimable rewards value)
- 'Wallet balance" - default tab, doesn't actually route. Wallet balance without earn nor rewards

Additionally, adds a new "Net worth" on top of those tabs with a reminder of the net worth that can be seen below on the dashboard balance chart.

- [x] mobile responsiveness, `Reward Balance` is cut off
- [x] include empty opportunities
- [x] flag?

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

### Prod - default `DefiDashboard` flag off

image
image
image

### This diff - `DashboardTabs` and `DefiDashboard` flags off

image
image
image

### This diff - flag on

image
image
image

### Responsiveness

image

### CODE REVIEW

File: src/pages/Dashboard/RewardsDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <Main headerComponent={<DashboardHeader />}>
+      <DeFiEarn
+        positionTableProps={{
+          filterBy: opportunities =>
+            opportunities.filter(opportunity => bnOrZero(opportunity.fiatRewardsAmount).gt(0)),
+        }}
+        providerTableProps={{
+          filterBy: opportunities =>
+            opportunities.filter(opportunity => bnOrZero(opportunity.fiatRewardsAmount).gt(0)),
+        }}
```
Comment (gomesalexandre) [inline src/pages/Dashboard/RewardsDashboard.tsx:18]: Could make this a pure `filter` callback fn and `Array.prototype.filter()` by it

File: src/pages/Dashboard/EarnDashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <Main headerComponent={<DashboardHeader />}>
+      <DeFiEarn
+        positionTableProps={{
+          filterBy: opportunities =>
+            opportunities.filter(opportunity => bnOrZero(opportunity.fiatAmount).gt(0)),
+        }}
+        providerTableProps={{
+          filterBy: opportunities =>
+            opportunities.filter(opportunity => bnOrZero(opportunity.fiatAmount).gt(0)),
+        }}
```
Comment (gomesalexandre) [inline src/pages/Dashboard/EarnDashboard.tsx:18]: We should use the right selector to diff. between the earn dashboard and rewards dashboard, not filtering like so

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4111** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4111" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4111?utm_source=stack-comment).

File: src/components/StakingVaults/ProviderTable.tsx [hunk 3]
```diff
@@ -62,14 +62,25 @@ const ResultsEmpty = ({ searchQuery }: { searchQuery?: string }) => {
   )
 }
 
-type PositionTableProps = {
+export type ProviderTableProps = {
```
Comment (gomesalexandre) [inline src/components/StakingVaults/ProviderTable.tsx:65]: revert

File: src/components/StakingVaults/PositionTable.tsx [hunk 4]
```diff
@@ -73,15 +73,26 @@ const ResultsEmpty = ({ searchQuery }: { searchQuery?: string }) => {
   )
 }
 
-type PositionTableProps = {
+export type PositionTableProps = {
```
Comment (gomesalexandre) [inline src/components/StakingVaults/PositionTable.tsx:76]:
```suggestion
type PositionTableProps = {
```

File: src/components/StakingVaults/ProviderTable.tsx (hunk 3 above)
Comment (gomesalexandre) [inline src/components/StakingVaults/ProviderTable.tsx:65]:
```suggestion
type ProviderTableProps = {
```

**Review by 0xdef1cafe — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**